vector/src/main/java/im/vector/app/features/voicebroadcast/listening/usecase/GetLiveVoiceBroadcastChunksUseCase.kt

```diff
@@ -124,12 +135,19 @@ class GetLiveVoiceBroadcastChunksUseCase @Inject constructor(
     /**
      * Transform the list of [TimelineEvent] to a mapped list of [MessageAudioEvent] related to a given voice broadcast.
      */
-    private fun List<TimelineEvent>.mapToChunkEvents(voiceBroadcastId: String, senderId: String?): List<MessageAudioEvent> =
+    private fun List<TimelineEvent>.mapToChunkEvents(voiceBroadcastId: String, senderId: String?): List<Event> =
             this.mapNotNull { timelineEvent ->
-                timelineEvent.root.asMessageAudioEvent()
-                        ?.takeIf {
-                            it.isVoiceBroadcast() && it.getVoiceBroadcastEventId() == voiceBroadcastId &&
-                                    it.root.senderId == senderId
-                        }
+                val event = timelineEvent.root
+                val relationContent = event.getRelationContent()
+                when {
+                    event.getClearType() == EventType.MESSAGE -> {
+                        event.asMessageAudioEvent()
+                                ?.takeIf {
+                                    it.isVoiceBroadcast() && it.getVoiceBroadcastEventId() == voiceBroadcastId && it.root.senderId == senderId
+                                }?.root
+                    }
+                    event.getClearType() == EventType.ENCRYPTED && relationContent?.type == RelationType.REFERENCE -> event
```

Comment:
I think we should also verify if the `eventId` in the reference is equal to the `voiceBroadcastId`